components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/AccessTokenIssuer.java

```diff
@@ -533,6 +534,20 @@ public OAuth2AccessTokenRespDTO issue(OAuth2AccessTokenReqDTO tokenReqDTO)
             }
         }
 
+        List<AccessTokenResponseHandler> tokenResponseHandlers = OAuthComponentServiceHolder.getInstance().
+                getAccessTokenResponseHandlers();
+        // Engaging token response handlers.
+        for (AccessTokenResponseHandler tokenResponseHandler : tokenResponseHandlers) {
+            Map<String, String> additionalTokenResponseAttributes =
+                    tokenResponseHandler.getAdditionalTokenResponseAttributes(tokReqMsgCtx);
+            if (additionalTokenResponseAttributes != null) {
+                for (String attribute : additionalTokenResponseAttributes.keySet()) {
+                    String attributeVal = additionalTokenResponseAttributes.get(attribute);
+                    tokenRespDTO.addParameter(attribute, attributeVal);
+                }
```

Comment:
Let's loop the entry set here so that we can avoid this attribute value lookup in each iteration again.